src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1119,6 +1137,7 @@ public UpdateTimestampsCache UpdateTimestampsCache
 		public ICache GetSecondLevelCacheRegion(string regionName)
 #pragma warning restore 618
 		{
+			CheckNotDisposed();
 			_allCacheRegions.TryGetValue(regionName, out var result);
 			return result;
```

Comment:
`_allCacheRegions` is also accessed by `private GetCache`, which I do not guard because it is currently called only by already directly guarded methods or upon initialization of the session factory.